packages/platform-core/src/components/NumberInput/NumberInputField.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FC, useCallback, useRef, useMemo } from 'react';
+import { cx } from 'emotion';
+import { useTheme } from '@grafana/ui';
+import { getStyles } from './NumberInput.styles';
+import { Field, FieldMetaState, FieldInputProps, UseFieldConfig } from 'react-final-form';
+import { Validator, compose } from '../../shared/validators';
+
+export interface NumberInputFieldProps extends UseFieldConfig<number> {
+  label?: string;
+  className?: string;
+  disabled?: boolean;
+  fieldClassName?: string;
+  name: string;
+  onChange?: (value: string) => undefined;
+  placeholder?: string;
+  required?: boolean;
+  showErrorOnBlur?: boolean;
+  validators?: Validator[];
+}
+
+interface NumberFieldRenderProps {
+  input: FieldInputProps<number>;
+  meta: FieldMetaState<number>;
+}
+
+export const NumberInputField: FC<NumberInputFieldProps> = React.memo(({
+  showErrorOnBlur = false,
+  className,
+  disabled = false,
+  fieldClassName,
+  label,
+  name,
+  placeholder,
+  required = false,
+  validators,
+  ...fieldConfig
+}) => {
+  const theme = useTheme();
+  const styles = useMemo(() => getStyles(theme), [theme]);
+  const inputId = `input-${name}-id`;
+  const validate = useMemo(() => (Array.isArray(validators) ? compose(...validators) : undefined), [
+    validators,
+  ]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const dispatchChangeEvent = useCallback(() => {
+    const event = new Event('change', { bubbles: true });
+
+    inputRef.current!.dispatchEvent(event);
```

Comment:
Isn't a property guard `?` a better choice?
```suggestion
    inputRef.current?.dispatchEvent(event);
```

Reply:
In case of applying this change, let's make it uniform further down as well.

Reply:
Done for all the three of them